pcsx2/PINE.cpp

```diff
@@ -406,32 +399,32 @@ PINEServer::IPCBuffer PINEServer::ParseCommand(char* buf, char* ret_buffer, u32
 			}
 			case MsgSaveState:
 			{
-				if (!m_vm->HasActiveMachine())
+				if (!VMManager::HasValidVM())
 					goto error;
 				if (!SafetyChecks(buf_cnt, 1, ret_cnt, 0, buf_size))
 					goto error;
-				StateCopy_SaveToSlot(FromArray<u8>(&buf[buf_cnt], 0));
+				VMManager::SaveStateToSlot(FromArray<u8>(&buf[buf_cnt], 0));
 				buf_cnt += 1;
 				break;
 			}
 			case MsgLoadState:
 			{
-				if (!m_vm->HasActiveMachine())
+				if (!VMManager::HasValidVM())
 					goto error;
 				if (!SafetyChecks(buf_cnt, 1, ret_cnt, 0, buf_size))
 					goto error;
-				StateCopy_LoadFromSlot(FromArray<u8>(&buf[buf_cnt], 0), false);
+				VMManager::LoadStateFromSlot(FromArray<u8>(&buf[buf_cnt], 0));
 				buf_cnt += 1;
 				break;
 			}
 			case MsgTitle:
 			{
-				if (!m_vm->HasActiveMachine())
+				if (!VMManager::HasValidVM())
 					goto error;
-				char* title = new char[GameInfo::gameName.size() + 1];
-				sprintf(title, "%s", GameInfo::gameName.ToUTF8().data());
-				const u32 size = strlen(title) + 1;
-				title[size] = 0x00;
+				const std::string& gameName = VMManager::GetGameName();
```

Comment:
This shouldn't be a reference, `GetGameName()` returns a new string object (that way it's thread-safe).